x-pack/metricbeat/module/iis/application_pool/application_pool.go

```diff
@@ -55,22 +55,25 @@ func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
 		log:           logp.NewLogger("application pool"),
 		reader:        reader,
 	}
-	if err := ms.reader.initCounters(config.Names); err != nil {
-		return ms, err
-	}
+
 	return ms, nil
 }
 
 // Fetch methods implements the data gathering and data conversion to the right
 // format. It publishes the event which is then forwarded to the output. In case
 // of an error set the Error field of mb.Event or simply call report.Error().
 func (m *MetricSet) Fetch(report mb.ReporterV2) error {
-	var config Config
-	if err := m.Module().UnpackConfig(&config); err != nil {
-		return nil
+	// refresh performance counter list
+	// Some counters, such as rate counters, require two counter values in order to compute a displayable value. In this case we must call PdhCollectQueryData twice before calling PdhGetFormattedCounterValue.
+	// For more information, see Collecting Performance Data (https://docs.microsoft.com/en-us/windows/desktop/PerfCtrs/collecting-performance-data).
+	// A flag is set if the second call has been executed else refresh will fail (reader.executed)
```

Comment:
Code added to `newReader` seems to be the same as a refresh, and it is run before `reader.executed`, could it also fail?

Reply:
we try to run one refresh at a time, running them one after another seems to yield exceptions